src/shared/mount-util.c

```diff
@@ -730,6 +730,46 @@ int umount_verbose(
         return 0;
 }
 
+int mount_beneath_graceful(int fsmount_fd, const char *dest, bool mount_beneath) {
```

Comment:
Bikeshedding: since this takes a `mount_beneath` parameter, maybe call it `mount_exchange_graceful`? Or it is somewhat confusing.